manifests/controllerconfig.crd.yaml

```diff
@@ -1116,8 +1116,25 @@ spec:
                                 type: string
                             type: object
                           external:
-                            description: External contains settings specific to the
-                              generic External infrastructure provider.
+                            description: External contains settings specific to the generic External infrastructure provider.
+                            properties:
+                              cloudControllerManager:
+                                description: cloudControllerManager contains settings specific to the external Cloud Controller Manager (a.k.a. CCM or CPI). When omitted, new nodes will be not tainted and no extra initialization from the cloud controller manager is expected.
+                                properties:
+                                  state:
+                                    description: "state determines whether or not an external Cloud Controller Manager is expected to be installed within the cluster. https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager \n Valid values are \"External\", \"None\" and omitted. When set to \"External\", new nodes will be tainted as uninitialized when created, preventing them from running workloads until they are initialized by the cloud controller manager. When omitted or set to \"None\", new nodes will be not tainted and no extra initialization from the cloud controller manager is expected."
+                                    enum:
+                                      - ""
```

Comment:
I am assuming "" is default, but I know some of these enums have a default specified for good practice.

Reply:
@adriengentil what is your suggestion here? I think having External would be better, right?

Reply:
I think the goal is to reflect what is actually defined in in the infrastructure object, and from what I understand, there is a plan to migrate the MCO to `openshift/api`, so I wouldn't introduce any discrepancies in terms of behavior here.

The default is "", because the field is a plain string. We can add a default, but it will create a diff if we try generate it with some scripts from the infrastructure object.

Reply:
https://github.com/openshift/machine-config-operator/blob/0a4602e80576f1ac0df14eca9eb4dddef0ce57de/manifests/controllerconfig.crd.yaml#L436-L466

https://github.com/openshift/machine-config-operator/blob/0a4602e80576f1ac0df14eca9eb4dddef0ce57de/manifests/controllerconfig.crd.yaml#L785C9-L807

from what I see the default is not set when "" is in the enum.